include/flatbuffers/flatbuffers.h

```diff
@@ -169,7 +169,8 @@ template<typename T, typename IT> struct VectorIterator {
   }
 
   difference_type operator-(const VectorIterator &other) const {
-    return (data_ - other.data_) / IndirectHelper<T>::element_stride;
+    return (data_ - other.data_) /
+           static_cast<difference_type>(IndirectHelper<T>::element_stride);
```

Comment:
Would it make sense to instead make `element_stride` directly of this type, or introduce a `element_diff` there?

Reply:
Yes, I've thought at first about changing the `element_stride` to `std::ptrdiff_t` first (code diffs would be much smaller), but then I thought it could be considered a breaking API change. If you think that it makes more sense to change the `element_stride` type (the only non-offsetting use is in the `std::bsearch` call), I will gladly change it.

Reply:
If it is more correct for `element_stride` to be `ptrdiff_t` (i.e. that is common in the STL), then lets change it, I doubt there will be a ton of people accessing this directly. If it's more common for it to be unsigned, lets introduce a `element_diff` here. @vglavnyy

Reply:
I agree that changing the `elemnt_stride` to `ptrdiff_t` will be better solution.
Can we add a small test case for `VectorIterator` with `std::prev`, `std::advance`, `std::distance`?